extensions/guacamole-vault/modules/guacamole-vault-ksm/src/main/java/org/apache/guacamole/vault/ksm/secret/KsmSecretService.java

```diff
@@ -304,18 +325,101 @@ private String getConnectionGroupKsmConfig(
 
     }
 
-    @Override
-    public Map<String, Future<String>> getTokens(UserContext userContext, Connectable connectable,
-            GuacamoleConfiguration config, TokenFilter filter) throws GuacamoleException {
+    /**
+     * Returns true if user-level KSM configuration is enabled for the given
+     * Connectable, false otherwise.
+     *
+     * @param connectable
+     *     The connectable to check for whether user-level KSM configs are
+     *     enabled.
+     *
+     * @return
+     *     True if user-level KSM configuration is enabled for the given
+     *     Connectable, false otherwise.
+     */
+    private boolean isKsmUserConfigEnabled(Connectable connectable) {
 
-        Map<String, Future<String>> tokens = new HashMap<>();
-        Map<String, String> parameters = config.getParameters();
+        // If it's a connection, user-level config is enabled IFF the appropriate
+        // attribute is set to true
+        if (connectable instanceof Connection)
+            return KsmAttributeService.TRUTH_VALUE.equals(((Connection) connectable).getAttributes().get(
```

Comment:
The `Attributes` interface might be a better choice.

Reply:
For the check on line 344? I guess it probably wouldn't do much harm to do this even for ConnectionGroups.

Reply:
The check and the typecast, yep.